nessie/s3_gateway_test.go

```diff
@@ -104,3 +106,133 @@ func TestS3UploadAndDownload(t *testing.T) {
 		})
 	}
 }
+
+func TestS3CopyObject(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+	destRepo := createRepositoryByName(ctx, t, "tests3copyobjectdest")
+
+	accessKeyID := viper.GetString("access_key_id")
+	secretAccessKey := viper.GetString("secret_access_key")
+	endpoint := viper.GetString("s3_endpoint")
+	opts := minio.PutObjectOptions{}
+	rand := rand.New(rand.NewSource(17))
+
+	creds := sigs[0].GetCredentials(accessKeyID, secretAccessKey, "")
+
+	type Object struct {
+		SourcePath, DestPath, Content string
+	}
+
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  creds,
+		Secure: false,
+	})
+	if err != nil {
+		t.Fatalf("minio.New: %s", err)
+	}
+
+	o := Object{
+		Content:    testutil.RandomString(rand, randomDataContentLength),
+		SourcePath: prefix + "source-file",
+		DestPath:   prefix + "dest-file",
+	}
+
+	_, err = minioClient.PutObject(
+		ctx, repo, o.SourcePath, strings.NewReader(o.Content), int64(len(o.Content)), opts)
+	if err != nil {
+		t.Errorf("minio.Client.PutObject(%s): %s", o.SourcePath, err)
+	}
+	// copy object to the same repository- update metadata to refer to the source address
+	_, err = minioClient.CopyObject(ctx,
+		minio.CopyDestOptions{
+			Bucket: repo,
+			Object: o.DestPath},
+		minio.CopySrcOptions{
+			Bucket: repo,
+			Object: o.SourcePath})
+
+	if err != nil {
+		t.Errorf("minio.Client.CopyObjectFrom(%s)To(%s): %s", o.SourcePath, o.DestPath, err)
+	}
+
+	download, err := minioClient.GetObject(
+		ctx, repo, o.DestPath, minio.GetObjectOptions{})
+	if err != nil {
+		t.Errorf("minio.Client.GetObject(%s): %s", o.DestPath, err)
+	}
+	// compere files content
+	contents := bytes.NewBuffer(nil)
+	_, err = io.Copy(contents, download)
+	if err != nil {
+		t.Errorf("download %s: %s", o.DestPath, err)
```

Comment:
```suggestion
		t.Fatal("download %s: %s", o.DestPath, err)
```
You could use Fatal in cases of errors where you don't want your test to continue running